tt_metal/common/core_coord.h

```diff
@@ -64,6 +64,93 @@ inline CoreCoord get_core_coord_from_relative(const RelativeCoreCoord &in, const
     return coord;
 }
 
+inline std::vector<CoreCoord> grid_to_cores(
```

Comment:
These methods each have its own implementation.
Can't we implement one most flexible option and others via that version instead?

Reply:
These are not new functions that I just added, I just moved them around in the file and outside the scope of this pr